### PR TITLE
Adding collision energy in precursor.

### PR DESCRIPTION
**Pull request by vijay-gnanasambandan-bruker**
Hi @swillems ,

I have made the necessary modifications to the precursor object by adjusting the collision energy. Kindly take a moment to review and provide any feedback or suggestions for further changes.

Thanks,
-Vijay

### CODE REVIEW

File: src/file_readers/spectrum_readers/dda_reader/precursors.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ impl PrecursorReader {
         let precursor_table: PrecursorTable =
             PrecursorTable::from_sql(&tdf_reader.tdf_sql_reader);
         let retention_times: Vec<f64> = tdf_reader.frame_table.rt.clone();
+        let collision_energy = &pasef_frames.collision_energy_by_precursor;
```
Comment (sander-willems-bruker) [inline src/file_readers/spectrum_readers/dda_reader/precursors.rs:32]: Make this plural `collision_energies` to improve readability

File: src/file_readers/common/sql_reader/tables/pasef_frame_msms.rs [hunk 2]
```diff
@@ -28,6 +29,7 @@ impl ReadableFromSql for PasefFrameMsMsTable {
                 .read_column_from_table("CollisionEnergy", table_name),
             precursor: sql_reader
                 .read_column_from_table("Precursor", table_name),
+            collision_energy_by_precursor: sql_reader.get_data_from_sql(&"select CollisionEnergy from PasefFrameMsMsInfo group by Precursor".to_string()),
```
Comment (sander-willems-bruker) [inline src/file_readers/common/sql_reader/tables/pasef_frame_msms.rs:32]: I think this conflicts with the setup of this class. It should just use the `read_from_table` structure to ensure the length of the array and the order is consistent with the rest of the table. Parsing it for a specific use case should happen at the appropriate spot, not here.

File: src/file_readers/spectrum_readers/dda_reader/precursors.rs (hunk 1 above)
Comment (sander-willems-bruker) [inline src/file_readers/spectrum_readers/dda_reader/precursors.rs:32]: This is probably yhe place to do `group by precursor` you currently do in the table loading.

File: src/file_readers/common/sql_reader.rs [hunk 3]
```diff
@@ -12,21 +12,27 @@ pub struct SqlReader {
...
     ) -> Vec<T> {
-        let connection: Connection = get_sql_connection(&self.path);
         let column_names: Vec<String> =
             self.get_table_columns(table_name).unwrap();
         let order_by: String = column_names.join(", ");
         let query: String = format!(
             "SELECT {} FROM {} ORDER BY {}",
             column_name, table_name, order_by
         );
+
```
Comment (sander-willems-bruker) [inline src/file_readers/common/sql_reader.rs:27]: I like that you adhere to the boy scout rule here and leave things better than you found them. Good addition to improve code readability, thanks!

File: src/file_readers/common/sql_reader.rs [hunk 4]
```diff
@@ -12,21 +12,27 @@ pub struct SqlReader {
 }
 
 impl SqlReader {
-    fn read_column_from_table<
-        T: rusqlite::types::FromSql + std::default::Default,
-    >(
+    fn read_column_from_table<T: rusqlite::types::FromSql + Default>(
```
Comment (sander-willems-bruker) [inline src/file_readers/common/sql_reader.rs:15]: Very minimal change, check if if it formatted right by running `cargo fmt`. Normally thse kind sof changes are done automaticaly. (I evne set up my editor (VSCode with rust-analyzer extension) to run `cargo fmt` automatically on saving)

**Comment by sander-willems-bruker** [comment]
Test seem to be failing, we need to add collision enegries to them, or even add a default to simplify things...

**Comment by vijay-gnanasambandan-bruker** [comment]
> Test seem to be failing, we need to add collision enegries to them, or even add a default to simplify things...

 Added default value in the Precursor object.  test working now.

**Pull request merged**